src/v/cluster/types.h

```diff
@@ -1276,7 +1280,8 @@ struct topic_properties
           batch_max_bytes,
           retention_local_target_bytes,
           retention_local_target_ms,
-          remote_delete);
+          remote_delete,
+          segment_ms);
```

Comment:
You might want to think about the ADL encoding of this type. Topic creation requests are forwarded to the leader, and, if you're unlucky enough for that to be the same request between the two nodes, it will be encoded using ADL. I think the default value of `nullopt` will be used for `segment_ms ` in this case. See `adl<cluster::topic_configuration>::from`.

We should've gotten rid of ADL ...